app/main.py

```diff
@@ -1,36 +1,58 @@
-from fastapi import FastAPI, Depends, HTTPException, status
+import sys
+import os
+
+# Add the parent directory to sys.path
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+
+from fastapi import FastAPI, Depends, HTTPException, status, Security
+from fastapi.security import OAuth2PasswordRequestForm
+from pydantic import EmailStr
+from typing import List, Annotated
 from sqlalchemy.orm import Session
-from app.database import SessionLocal, engine
-from app.models import Base, UserDB
-from app.schemas import UserCreate, UserOut
-from app.crud import create_user, get_user_by_email
+from database import engine, get_db, Base
+from models.user import UserDB
+from schemas.user import UserCreate, UserOut, User
+from schemas.task import TaskCreate, TaskOut, TaskStatus
+from schemas.token import Token
+from crud.user import create_user, get_user_by_email
+from crud.task import create_task, get_tasks, get_task, update_task, delete_task, get_tasks_by_priority, get_tasks_by_status, get_task_by_deadline
+from app.utils.auth import get_current_user, create_access_token, verify_password
+from datetime import datetime
 
-# Create all tables in the database
 Base.metadata.create_all(bind=engine)
 
 app = FastAPI()
 
-# Dependency to get DB session
-def get_db():
-    db = SessionLocal()
-    try:
-        yield db
-    finally:
-        db.close()
 
-@app.post("/users/", response_model=UserOut, status_code=status.HTTP_201_CREATED)
+@app.post("/register/", response_model=UserOut, status_code=status.HTTP_201_CREATED)
```

Comment:
So both register, login and task endpoints in the same file? You didnt fix the structure that I sent you, that is a top priority MUST BE implemented like that, this file now is something we call KAMARA